static/app/views/issueDetails/groupCheckIns.tsx

```diff
@@ -0,0 +1,311 @@
+                start: start.toLocaleString(),
+                end: ((currentCursor?.offset ?? 0) + pageCount).toLocaleString(),
+                // TODO(leander): See if we can get a total count from the backend
+              })}
+        </EventListHeaderItem>
+        <EventListHeaderItem>
+          <ButtonBar gap={0.25}>
+            <GrayLinkButton
+              aria-label={t('Previous Page')}
+              borderless
+              size="xs"
+              icon={<IconChevron direction="left" />}
+              to={{
+                ...location,
+                query: {
+                  ...location.query,
+                  cursor: links.previous?.cursor,
+                },
+              }}
+              disabled={isEventPending || previousDisabled}
+            />
+            <GrayLinkButton
+              aria-label={t('Next Page')}
+              borderless
+              size="xs"
+              icon={<IconChevron direction="right" />}
+              to={{
+                ...location,
+                query: {
+                  ...location.query,
+                  cursor: links.next?.cursor,
+                },
+              }}
+              disabled={isEventPending || nextDisabled}
+            />
+          </ButtonBar>
+        </EventListHeaderItem>
+      </EventListHeader>
+      <GridEditable
+        isLoading={isEventPending}
+        data={uptimeData}
+        columnOrder={[
+          {key: 'timestamp', width: COL_WIDTH_UNDEFINED, name: t('Timestamp')},
+          {key: 'checkStatus', width: 115, name: t('Status')},
+          {key: 'durationMs', width: 110, name: t('Duration')},
+          {key: 'environment', width: 115, name: t('Environment')},
+          {key: 'traceId', width: 100, name: t('Trace')},
+          {key: 'region', width: 100, name: t('Region')},
+          {key: 'uptimeCheckId', width: 100, name: t('ID')},
+        ]}
+        columnSortBy={[]}
+        grid={{
+          renderHeadCell: (col: GridColumnOrder) => <Cell>{col.name}</Cell>,
+          renderBodyCell: (column, dataRow) => (
+            <CheckInBodyCell
+              column={column}
+              dataRow={dataRow}
+              userOptions={user.options}
+            />
+          ),
+        }}
+      />
+    </StreamlineEventsTable>
+  );
+}
+
+function CheckInBodyCell({
+  dataRow,
+  column,
+  userOptions,
+}: {
+  column: GridColumnOrder<string>;
+  dataRow: UptimeCheckIn;
+  userOptions: User['options'];
+}) {
+  const theme = useTheme();
+  const columnKey = column.key as keyof UptimeCheckIn;
+  const cellData = dataRow[columnKey];
+
+  if (!cellData) {
+    return <Cell />;
+  }
+
+  switch (columnKey) {
+    case 'timestamp':
+      const format = userOptions.clock24Hours
+        ? 'MMM D, YYYY HH:mm:ss z'
+        : 'MMM D, YYYY h:mm:ss A z';
+      return (
+        <Cell>
+          <TimeSince
+            tooltipShowSeconds
+            unitStyle="short"
+            date={new Date(cellData)}
```

Comment:
nit would prefer the string? here since the date object would potentially cause more rerenders
```suggestion
            date={cellData}
```